warehouse/internal/loadfiles/loadfiles_test.go

```diff
@@ -484,6 +484,283 @@ func TestGetLoadFilePrefix(t *testing.T) {
 	}
 }
 
+func TestGroupStagingFiles(t *testing.T) {
+	t.Run("size based grouping", func(t *testing.T) {
+		testCases := []struct {
+			name          string
+			files         []*model.StagingFile
+			batchSizes    []int
+			skipSizeCheck bool
+		}{
+			{
+				name:       "empty files",
+				files:      []*model.StagingFile{},
+				batchSizes: []int{},
+			},
+			{
+				name: "single file under limit",
+				files: []*model.StagingFile{
+					{
+						BytesPerTable: map[string]int64{
+							"table1": 50 * 1024 * 1024, // 50MB
+						},
+					},
+				},
+				batchSizes: []int{1},
+			},
+			{
+				name: "multiple files over limit",
+				files: []*model.StagingFile{
+					{
+						BytesPerTable: map[string]int64{
+							"table1": 150 * 1024 * 1024, // 150MB
+						},
+					},
+					{
+						BytesPerTable: map[string]int64{
+							"table1": 150 * 1024 * 1024, // 150MB
+						},
+					},
+				},
+				batchSizes:    []int{1, 1},
+				skipSizeCheck: true,
+			},
+			{
+				name: "multiple files under limit",
+				files: []*model.StagingFile{
+					{
+						BytesPerTable: map[string]int64{
+							"table1": 30 * 1024 * 1024, // 30MB
+						},
+					},
+					{
+						BytesPerTable: map[string]int64{
+							"table1": 40 * 1024 * 1024, // 40MB
+						},
+					},
+				},
+				batchSizes: []int{2},
+			},
+			{
+				name: "multiple files over limit",
```

Comment:
There is another at line 512 with same tag `multiple files over limit`?

Reply:
Removed one of the `multiple files over limit` tests